.circleci/config.yml

```diff
@@ -127,29 +132,37 @@ jobs:
           command: |
             VARS="-var tags={\"build_url\":\"$CIRCLE_BUILD_URL\"}"
             VARS+=' -var launch_type=<<parameters.launch_type>>'
+            VARS+=' -var consul_version=<<parameters.consul_version>>'
             case $CIRCLE_BRANCH in
-                main | release/*) VARS+=" -var enable_hcp=true";;
+                main | release/*) VARS+=" -var enable_hcp=<<parameters.enable_hcp>>";;
                 *) VARS+=" -var enable_hcp=false";;
             esac
 
             terraform destroy -auto-approve $VARS
           when: always
 
+
+acceptance-common: &acceptance-common
+  filters:
+    branches:
+      # Forked pull requests have CIRCLE_BRANCH set to pull/XXX
+      ignore: /pull\/[0-9]+|docs\/.*/
+  requires:
+    - go-fmt-and-lint-acceptance
+
 workflows:
   version: 2.1
   test-and-build:
     jobs:
       - go-fmt-and-lint-acceptance
       - terraform-fmt
-      - acceptance:
-          matrix:
-            parameters:
-              launch_type:
-                - FARGATE
-                - EC2
-          filters:
-            branches:
-              # Forked pull requests have CIRCLE_BRANCH set to pull/XXX
-              ignore: /pull\/[0-9]+|docs\/.*/
-          requires:
-            - go-fmt-and-lint-acceptance
+      # We have a limit of 6 HCP Consul clusters.
+      # The following controls whether to enable HCP when testing release branches.
+      # HCP is always disabled for tests on PRs.
+      - acceptance: {name: "acceptance-1.12-FARGATE-HCP", consul_version: '1.12.6', enable_hcp: true, launch_type: FARGATE, <<: *acceptance-common}
```

Comment:
One thing that is confusing: The job will be named `acceptance-1.12-FARGATE-HCP`. When the job runs on release branches, it will test with HCP, but we never enable HCP for tests on PRs. 

I'm sure when someone else looks at the job name, they will think that it is running tests with HCP on their PR, which isn't the case.

Reply:
Yes, I noticed this in the pipeline but I think it is OK because we just don't test HCP on PRs period. I like the approach that you've taken here to be explicit about what the tests are doing and I think it is a good tradeoff.